modules/support-collecting-host-network-trace.adoc

```diff
@@ -0,0 +1,68 @@
+
+[id="support-collecting-host-network-trace_{context}"]
+= Collecting a host network trace
+
+Sometimes, troubleshooting a network-related issue is simplified by tracing network communication and capturing packets on multiple nodes at the same time.
+
+You can use a combination of the `oc adm must-gather` command and the `registry.redhat.io/openshift4/network-tools-rhel8` container image to gather packet captures from nodes.
+You can analyze the packet captures to understand network communication and assist with diagnosing network issues.
+
+The `oc adm must-gather` command is used to run the `tcpdump` command in pods on specific nodes.
+The `tcpdump` command records the packet captures in the pods.
+When the `tcpdump` command exits, the `oc adm must-gather` command transfers the files with the packet captures from the pods to your client machine.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Start a packet capture from the host network on some nodes:
++
+[source,terminal]
+----
+$ oc adm must-gather \
+    --dest-dir /tmp/captures \  <.>
+    --source-dir '/tmp/tcpdump/' \  <.>
+    --image registry.redhat.io/openshift4/network-tools-rhel8:latest \  <.>
+    --node-selector 'node-role.kubernetes.io/worker' \  <.>
+    --host-network=true \  <.>
+    --timeout 30s \  <.>
+    -- \
+    tcpdump -i any \  <.>
+    -w /tmp/tcpdump/\$POD_NAME-%Y-%m-%dT%H:%M:%S.pcap -W 1 -G 300  <.>
+----
+<.> The command stores the packet captures in directories that are relative to `/tmp/captures` on the client machine. You can specify any writable directory.
+<.> When `tcpdump` is run in the debug pod that `oc adm must-gather` starts, the packet captures are temporarily stored in the `/tmp/tcpdump` directory on the pod.
+<.> The specified image includes the `tcpdump` command that is needed to perform the packet captures.
+<.> The packet captures are performed on worker nodes because they match the specified node selector. As an alternative, you can specify the `--node-name` argument instead to run the packet capture on a single node. If you omit both the `--node-selector` and the `--node-name` argument, the packet captures are performed on all nodes.
+<.> The `--host-network=true` argument is required so that the packet captures are performed on the network interfaces of the node.
+<.> The debug pod is run with a timeout of 30 seconds. If you do not specify the `--timeout` argument and a duration, the debug pod runs for 10 minutes.
+<.> The `-i any` argument for the `tcpdump` interface specifies to capture packets on all network interfaces. As an alternative, you can specify a network interface name.
+<.> The sample command demonstrates one way to use the `POD_NAME` environment variable so that the filenames for the packet captures are unique. The `NODE_NAME` environment variable is also available for the same purpose.
```

Comment:
FYI: these are actually a bit superfluous since https://github.com/openshift/oc/pull/965 and they should really have been removed with that PR. Since that PR the directory output for the must-gather will look like (if running against multiple nodes). So the user doesn't actually need to use these variables to differentiate the files.

```
$ tree tmp
tmp
├── event-filter.html
├── ovn-control-plane
│   └── quay-io-openshift-origin-network-tools-sha256-e00b183ae55dc935b95e735957d8ec4d107fd39f365e1b45f01eeaf930fce0a8
│       └── 2021-11-18T10:21:48.pcap
├── ovn-worker
│   └── quay-io-openshift-origin-network-tools-sha256-e00b183ae55dc935b95e735957d8ec4d107fd39f365e1b45f01eeaf930fce0a8
│       └── 2021-11-18T10:21:48.pcap
├── ovn-worker2
│   └── quay-io-openshift-origin-network-tools-sha256-e00b183ae55dc935b95e735957d8ec4d107fd39f365e1b45f01eeaf930fce0a8
│       └── 2021-11-18T10:21:48.pcap
└── timestamp

6 directories, 5 files
```

Reply:
@alexanderConstantinescu  - with this directory structure, are we losing ability to differentiate which files corresponds to which nodes? if so, wouldn't that be something desired by users?

Reply:
> with this directory structure, are we losing ability to differentiate which files corresponds to which nodes? 

No, `ovn-worker`/`ovn-control-plane`/`ovn-worker2` are nodes in this cluster I used to capture the data.

Reply:
And in any case: those environment variables still exist. So a user is still free to use them if they need / want